src/components/PublishingStatus.tsx

```diff
@@ -23,34 +24,67 @@ function RelayPublishStatus({
   status: Array<PublishStatus>;
   relayUrl: string;
 }): JSX.Element {
-  const navigate = useNavigate();
+  const isMobile = useMediaQuery(IS_MOBILE);
+  const [showDetails, setShowDetails] = useState<boolean>(false);
   const numberFulfilled = getStatusCount(status, "fulfilled");
   const numberRejected = getStatusCount(status, "rejected");
   const totalNumber = numberFulfilled + numberRejected;
   const percentage = Math.round((numberFulfilled / totalNumber) * 100);
-  const isWarning = percentage < 50;
+  const isWarning = percentage < 80;
+  const warningVariant = percentage < 50 ? "danger" : "warning";
   const lastRejectedReason = getLastRejectedReason(status);
   return (
-    <>
+    <div style={{ maxWidth: "100vw" }}>
       <Dropdown.Divider />
-      <Dropdown.Item tabIndex={0} onClick={() => navigate("/relays")}>
-        <div className="flex-row-space-between">
+      <Dropdown.Item tabIndex={0}>
+        <div
+          role="button"
+          className="flex-row-space-between"
+          tabIndex={0}
+          onClick={(e) => {
+            e.stopPropagation();
+            setShowDetails(!showDetails);
+          }}
+          onKeyDown={(e) => {
+            if (e.key === "Enter") {
+              e.stopPropagation();
+              setShowDetails(!showDetails);
+            }
+          }}
+        >
           <div className="me-2">
-            <div className="bold">{`Relay ${relayUrl}:`}</div>
-            <div className="mt-1">
-              {totalNumber > 0
-                ? `${
-                    totalNumber === 1
-                      ? `The last event ${
-                          numberFulfilled === 1 ? "could" : "could not"
-                        }`
-                      : `${percentage}% of the last ${totalNumber} events could`
-                  } be published on this relay`
-                : `No events were attempted to be published on this relay`}
-            </div>
-            {lastRejectedReason && (
-              <div>
-                {`The last event could not be published because: ${lastRejectedReason}`}
+            <div className="bold break-word">{`Relay ${relayUrl}:`}</div>
+            <ProgressBar
+              now={percentage}
+              label={`${percentage}%`}
+              variant={isWarning ? warningVariant : "success"}
+              style={{
+                width: "30rem",
+                maxWidth: isMobile ? "70vw" : "30rem",
+                height: "1.5rem",
+              }}
+            />
+            {showDetails && (
```

Comment:
this bit is a bit too complex to use inline jsx, how about an own Collapse component? Would reduce lines of code and makes it way more readable

Reply:
i dont need the collapse at all...

Reply:
Still a bit complicated, pls. move it to a own component